README.md

```diff
@@ -1,3 +1,360 @@
+   malicious PECL package. Ideally, `npecl` would run without `sudo`, and just prompt for access, if possible.
+
+### CLI commands
+
+#### `install {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Installs the requested package. The package may be requested with or without `ext-` prefix. `npecl install ext-xdebug`
+and `npecl install xdebug` would be equivalent.
+
+If `version-constraint` is given, try to install that version if it matches the allowed versions.
+
+On Windows this downloads the correct DLL (attached as file to the release tag), if available. Otherwise, it will
+download the source and compile the extension.
+
+It will then attempt to create a 20-{extension_name}.ini file with `extension={extension_name}` in the directory
+returned by `php-config –ini-dir`, the name that is used is the one from the "name" element in the metadata file.
+
+If `@dev-branch-name` is given, try to install from the branch called "branch-name", for example use `@dev-master` to
+compile from your master branch. As branches would usually not have binaries, this would not work for Windows.
+
+You can't use both a `version-constraint` and `@dev-branch-name` at the same time.
+
+Each step should run with as few privileges as possible.
+
+#### `build {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as install, but only compiles (or downloads if it's Windows).
+
+#### `changelog {?version}`
+
+Shows the release notes of the version it was going to install, or from the specific version, if given.
+
+#### `download {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as build, but puts the files in a local directory for manual building and installation.
+
+#### `info`
+
+Shows the description from the metadata file
+
+#### `list`
+
+Shows all available commands
+
+#### `self-update`
+
+Attempts to upgrade the npecl tool itself
+
+#### `show`
+
+Shows all installed extensions available with the PHP version in the path, including their versions. This includes
+*all* loaded PHP extensions, and not just npecl ones.
+
+#### `verify {?version-constraint}`
+
+Shows the signature of the person who signed the version npecl would install, or from the specific version, if given.
+
+#### `upgrade {?{ext-name}{?:version-constraint}{?@dev-branch-name}}`
+
+Attempts to upgrade all installed versions to the latest available ones on GitHub (unless their major version has
+changed) If a specific package is specified, only attempt to upgrade that specific one.
+
+### CLI Options
+
+#### `--dry-run`
+
+Shows what it is about to do, but doesn't actually install
+
+#### `--force`
+
+To attempt to install a version that doesn't match the version constraints from the meta-data, for instance to install
+an older version than recommended, or when the signature is not available.
+
+#### `--pre-release`
+To allow for installation of pre-release versions
```

Comment:
I'm removing this, since we can specify pre-release versions using Composer's version constraints anyway

Reply:
Removed